agent/immortalstreams/backedpipe/ring_buffer_internal_test.go

```diff
@@ -0,0 +1,264 @@
+package backedpipe
+
+import (
+	"bytes"
+	"os"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestMain(m *testing.M) {
+	if runtime.GOOS == "windows" {
+		// Don't run goleak on windows tests, they're super flaky right now.
+		// See: https://github.com/coder/coder/issues/8954
+		os.Exit(m.Run())
```

Comment:
If Windows CLI tests are still flaky, then we need to look into it. That issue is closed not planned and has been for years.